src/main/groovy/com/bmuschko/gradle/docker/DockerJavaApplication.groovy

```diff
@@ -17,33 +17,29 @@ package com.bmuschko.gradle.docker
 
 import com.bmuschko.gradle.docker.tasks.image.Dockerfile
 import org.gradle.api.Action
+import org.gradle.api.Project
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.Nested
 
 class DockerJavaApplication {
```

Comment:
Can you annotate this with `@CompileStatic` now?

Reply:
I was planning to make everything in this code base statically compilable...at least as much as possible. However, I want to tackle this as part of a different PR. This PR is already enormous.